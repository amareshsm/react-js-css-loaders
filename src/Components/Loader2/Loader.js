import React from 'react'
import styles from './style.css'

const Loader2 = (props) => (
  <>
    <style>{`${styles}`}</style>
    <div className="loader-div two">
      <div className="loader2">
        <div className="loader-ring">
          <div></div>
          <div></div>
          <div></div>
          <div></div>
        </div>
      </div>
    </div>
  </>
)

export default Loader2
