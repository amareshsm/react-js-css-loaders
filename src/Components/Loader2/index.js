export { default as Revolver } from './Loader'
