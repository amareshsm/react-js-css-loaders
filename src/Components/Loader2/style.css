.loader2 {
    width: 100px;
    height: 100px;
}

.loader2 .loader-ring {
    display: inline-block;
    position: relative;
    width: 95px;
    height: 95px;
}

.loader2 .loader-ring div {
    display: block;
    position: absolute;
    width: 60px;
    height: 60px;
    border: 6px solid black;
    border-radius: 50%;
    border-color: black transparent transparent transparent;
    animation: spin 2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
}

@keyframes spin {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
}
