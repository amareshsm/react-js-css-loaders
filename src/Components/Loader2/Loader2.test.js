import React from 'react'
import renderer from 'react-test-renderer'
import { Loader2 } from '../Loader2'
it('renders correctly', () => {
  const tree = renderer.create(<Loader2 />).toJSON()
  expect(tree).toMatchSnapshot()
})
