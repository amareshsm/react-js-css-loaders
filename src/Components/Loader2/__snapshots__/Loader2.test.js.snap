// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <style>
    [object Object]
  </style>,
  <div
    className="loader-div two"
  >
    <div
      className="loader2"
    >
      <div
        className="loader-ring"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>,
]
`;
